File: src/main/java/it/unitn/dbtrento/spark/utils/InputFormat.java
package it.unitn.dbtrento.spark.utils;

public enum InputFormat {
  CSV, PARQUET, TSV, SSV;

  public char getFieldDelimiter() {
    switch (this) {
      case CSV:
        return FieldDelimiter.COMMA.getFieldDelimiter();
      case PARQUET:
        return FieldDelimiter.COMMA.getFieldDelimiter();
      case TSV:
        return FieldDelimiter.TAB.getFieldDelimiter();
      case SSV:
        return FieldDelimiter.SEMICOLON.getFieldDelimiter();
      default:
        System.err.println("Error while getting the field delimiter.");
        return ',';
    }
  }
}

File: src/main/scala/eu/unitn/disi/db/spark/utils/InputFormat.scala
package eu.unitn.disi.db.spark.utils

import org.slf4j.{Logger, LoggerFactory}

object InputFormat extends Enumeration {

  val log : Logger = LoggerFactory.getLogger(this.getClass.getName)

  val CSV, PARQUET, TSV, SSV, JSON = Value

  implicit def getFieldDelimiter(format: Value): Char = format match {
    case CSV =>
      FieldDelimiter.COMMA.getFieldDelimiter
    case PARQUET =>
      FieldDelimiter.COMMA.getFieldDelimiter
    case TSV =>
      FieldDelimiter.TAB.getFieldDelimiter
    case SSV =>
      FieldDelimiter.SEMICOLON.getFieldDelimiter
    case _ =>
      log.error("Error while getting the field delimiter.")
      ','
  }
}
